compaction/compaction_manager.cc

```diff
@@ -1147,7 +1147,13 @@ void compaction_manager::do_stop() noexcept {
 }
 
 inline bool compaction_manager::can_proceed(table_state* t) const {
-    return (_state == state::enabled) && _compaction_state.contains(t) && !_compaction_state.at(t).compaction_disabled();
+    if (_state != state::enabled) {
+        return false;
+    }
+    if (auto cs = _compaction_state.find(t); cs != _compaction_state.end()) {
+        return !cs->second.compaction_disabled();
+    }
+    return true;
```

Comment:
If state != enable and compaction_state.contains returns false, then old code would result in false, while new one would get here and return true

Reply:
@xemul thanks. probably what you meant was "If state == enabled". but yeah, i misread the existing code. should be better now.